### ISSUE TITLE
[BUG]: TypeScript throws errors when adding [key: string]: string if other properties are not of type string

### ISSUE THREAD

**Issue by RodrigoTomeES** [microsoft/TypeScript#58194]
### 🔎 Search Terms

[key: string]: string override other types

### 🕗 Version & Regression Information

- This is an error with types

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=5.4.5#code/C4TwDgpgBAKhDOwoF4oG8BQUrAcAXOlAK4BOANoYqQJYB2A5lAL4DcWUA2gNYQhXBajALoChDdmyA

### 💻 Code

```ts
type Test = {
  test: { url: string };
  [key: string]: string;
};
```

### 🙁 Actual behavior

Throw error in test because is not a string

### 🙂 Expected behavior

Doesn't throw an error

**Comment by MartinJohns** [comment]
This is working as intended. The type of the index signature must be at least a union including all types of the known properties.

If you **really** want this, you can use an intersection type instead: `{ test: { url: string } } & { [key: string]: string }`  
But be aware that this is **not sound**. It's easy to still assign a `string` to the property `test`.

**Comment by fatcerberus** [comment]
To represent this type properly you'd need https://github.com/microsoft/TypeScript/issues/17867
